src/encoding/utf32le_encoding_object.cpp

```diff
@@ -0,0 +1,88 @@
+#include "natalie.hpp"
+#include <stdio.h>
+
+namespace Natalie {
+
+std::pair<bool, StringView> Utf32LeEncodingObject::prev_char(const String &string, size_t *index) const {
+    if (*index == 0)
+        return { true, StringView() };
+
+    if (*index < 4) {
+        (*index)--;
+        return { false, StringView(&string, *index, 1) };
+    }
+
+    size_t i = *index - 4;
+
+    // check codepoint
+    StringView character = StringView(&string, i, 4);
+    nat_int_t codepoint = decode_codepoint(character);
+    if (codepoint < 0 || codepoint > 0x10FFFF || (codepoint >= 0xD800 && codepoint <= 0xDFFF)) {
+        (*index)++;
+        return { false, StringView(&string, i) };
+    }
+
+    *index -= 4;
+    return { true, StringView(&string, i, 4) };
+}
```

Comment:
Looks like this method isn't tested. What is a proper place for such test?

Reply:
I think it's used by String#chomp

Reply:
We could add some tests for `String#chomp` with different encodings in string_test.rb.

Reply:
Done